std/getopt.d

```diff
@@ -695,20 +713,22 @@ private bool handleOption(R)(string option, R receiver, ref string[] args,
                 val = args[i];
                 args = args[0 .. i] ~ args[i + 1 .. $];
             }
-            static if (is(typeof(*receiver) == enum))
+            static if (is(typeof(*receiver) == enum) ||
+                is(typeof(*receiver) == string))
             {
-                *receiver = to!(typeof(*receiver))(val);
+                handleConversion(option, val, receiver, i);
             }
             else static if (is(typeof(*receiver) : real))
             {
                 // numeric receiver
-                if (incremental) ++*receiver;
-                else *receiver = to!(typeof(*receiver))(val);
-            }
-            else static if (is(typeof(*receiver) == string))
-            {
-                // string receiver
-                *receiver = to!(typeof(*receiver))(val);
+                if (incremental)
+                {
+                    ++*receiver;
+                }
+                else
+                {
+                    handleConversion(option, val, receiver, i);
+                }
```

Comment:
While I agree that the original code is hard to read, expanding it into 3-line braced blocks seems excessive. What about just leaving out those braces?